packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -6,41 +6,53 @@ import Typography from '@material-ui/core/Typography';
 import Collapse from '@material-ui/core/Collapse';
 import { withStyles } from '@material-ui/core/styles';
 import { useForkRef } from '@material-ui/core/utils';
+import { useIsFocusVisible } from '../../../material-ui/src/utils/focusVisible';
 import TreeViewContext from '../TreeView/TreeViewContext';
 
-export const styles = {
-  /* Styles applied to the root element. */
-  root: {
-    listStyle: 'none',
-    margin: 0,
-    padding: 0,
-    outline: 0,
-    '&:focus > $content': {
-      outline: 'auto 1px',
+export const styles = function styles(theme) {
+  return {
+    /* Styles applied to the root element. */
+    root: {
+      listStyle: 'none',
+      margin: 0,
+      padding: 0,
+      outline: 0,
+      cursor: 'pointer',
     },
-  },
-  /* Styles applied to the `role="group"` element. */
-  group: {
-    margin: 0,
-    padding: 0,
-    marginLeft: 26,
-  },
-  /* Styles applied to the tree node content. */
-  content: {
-    width: '100%',
-    display: 'flex',
-    alignItems: 'center',
-  },
-  /* Styles applied to the tree node icon and collapse/expand icon. */
-  iconContainer: {
-    marginRight: 2,
-    width: 24,
-    minWidth: 24,
-    display: 'flex',
-    justifyContent: 'center',
-  },
-  /* Styles applied to the label element. */
-  label: {},
+    /* Styles applied to the `role="group"` element. */
+    group: {
+      margin: 0,
+      padding: 0,
+      marginLeft: 26,
+    },
+    /* Styles applied to the tree node content. */
+    content: {
+      width: '100%',
+      display: 'flex',
+      alignItems: 'center',
+      '&:focus': {
+        outline: 'none',
+      },
+      '&$focusVisible': {
```

Comment:
I was wondering about this point too when reviewing the component. I have seen Josh implement it and revert it when iterating. I think that the reasoning is: wcag doesn't make any distinction, https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-2/treeview-2a.html, Finder on macOS doesn't make any distinction, Windows and the folder explorer doesn't make any distinction.

Reply:
Yeah, I removed it when I realised wcag didn’t make a distinction

Reply:
Hmm, you're right. My mental model of a tree was that the nodes didn't hold visible focus, but that isn't correct. I'll drop that commit.

Reply:
In that case we can probably drop the onClick handler from the demo, as the focus indicator is enough to show a leaf node has been clicked.